crates/nu-cli/src/completions/completer.rs

```diff
@@ -56,13 +57,90 @@ impl NuCompleter {
         suggestions
     }
 
+    fn external_completion(
+        &self,
+        decl_id: usize,
+        line: String,
+        pos: usize,
+        offset: usize,
+    ) -> Vec<Suggestion> {
+        let span = Span {
+            start: offset,
+            end: pos,
+        };
+
+        let line_pos = pos - offset;
+        let stack = self.stack.clone();
+
+        let block = self.engine_state.get_block(decl_id);
+        let mut callee_stack = stack.gather_captures(&block.captures);
+
+        for (idx, PositionalArg { var_id, .. }) in
```

Comment:
Instead of using a for loop, why don't you use get?

Reply:
true